udata_front/theme/gouvfr/assets/js/components/search/OrganizationSearch/OrganizationSearch.vue

```diff
@@ -94,68 +77,23 @@ async function fetchOptions() {
   } catch (error) {
     console.error('Error fetching options:', error);
   }
-}
+};
 
 const input = ref<HTMLElement | null>(null);
-const button = ref<HTMLElement | null>(null);
 const list = ref<HTMLElement | null>(null);
 
-function showAndFocus () { 
-  if (!expanded.value) {
-    input.value?.focus();
-    showAndSelectIfQuery();
-  } else {
-    searchSelectedOption();
-  }
-}
-
-function showAndSelectIfQuery() {
-  if (q.value && options.value.length > 0) {
-    show();
-    selectedIndex.value = 0;
-  }
-}
-
-function handleKeyDown(e: KeyboardEvent) {
-  switch (e.key) {
-    case "ArrowDown":
-      e.preventDefault();
-      selectNextOption();
-      break;
-    case "ArrowUp":
-      e.preventDefault();
-      selectPreviousOption();
-      break;
-    case "Enter":
-      e.preventDefault();
-      searchSelectedOption();
-      break;
-    default:
-      showAndSelectIfQuery();
-      break;
-  }
-};
-
-function handleFocusOut() {
-  hide();
-};
+onMounted(() => registerBackgroundEvent(input, list, null));
```

Comment:
Can you move the `onMounted`, `onUnmounted` and the refs to the composable too ?